# origin: horms (BitKeeper)
# cset: 1.1530 (2.4) key=41c7fcec2_zU8vXOtj5_oVAvgUVI0w
# inclusion: upstream
# descrition: [PATCH] binfmt_elf force_sig arguments fix
# revision date: Wed, 22 Dec 2004 11:10:36 +0900
#
# S rset: ChangeSet|1.1529..1.1530
# I rset: fs/binfmt_elf.c|1.36..1.37
#
# This is a BitKeeper generated diff -Nru style patch.
#
# fs/binfmt_elf.c
#   2004/12/21 10:21:25-02:00 horms@verge.net.au +1 -1
#   binfmt_elf force_sig arguments fix
# 
# ChangeSet
#   2004/12/21 08:37:32-02:00 horms@verge.net.au 
#   [PATCH] binfmt_elf force_sig arguments fix
#   
#   There appears to be a small error in the change that was recently
#   applied to fs/binfmt_elf.c to fix error codes and eraly corrupt
#   binary detection.
#   
#   The patch includes changing a send_sig() call to a force_sig() call in
#   load_elf_binary(). However force_sig() only accepts 2 arguments, and
#   thus the patch causes the build to fail.
#   
#   I propose the following patch to simply remove the extra argument to
#   force_sig(), which I beleive will give a sensible result.  That or
#   change the call back to send_sig(), though I assume it was changed to
#   force_sig() for a reason.
# 
#
===== fs/binfmt_elf.c 1.36 vs 1.37 =====
--- 1.36/fs/binfmt_elf.c	2004-12-18 03:17:46 +09:00
+++ 1.37/fs/binfmt_elf.c	2004-12-21 21:21:25 +09:00
@@ -806,7 +806,7 @@
 		if (BAD_ADDR(elf_entry)) {
 			printk(KERN_ERR "Unable to load interpreter %.128s\n",
 				elf_interpreter);
-			force_sig(SIGSEGV, current, 0);
+			force_sig(SIGSEGV, current);
 			retval = -ENOEXEC; /* Nobody gets to see this, but.. */
 			goto out_free_dentry;
 		}
